### ISSUE TITLE
fieldNamed does not work for columns that refetence other tables

### ISSUE THREAD

**Issue by Fingerzam** [haskell-beam/beam#579]
The documentation instructs using `Database.Beam.Schema.Tables.fieldNamed` if the default name does not match actual column name. However, if I try to use it for a column that references another table, it fails to compile with

```
    • Couldn't match expected type ‘PrimaryKey
                                      RunT
                                      (BeamTables.FieldModification
                                         (BeamTables.TableField DbStateT))’
                  with actual type ‘BeamTables.FieldModification
                                      (BeamTables.TableField tbl0) a0’

```

Is there another function that I'm supposed to use here? If so, could it be added to the documentation? Otherwise, can such a function be added together with an IsString instance?

**Comment by alexfmpe** [comment]
Going by similar looking 'nesting' issues I've ran into many time with beam, I'd assume you need something like `YourTypesPrimaryKeyConstructor (fieldNamed "foo")`

**Comment by kmicklas** [comment]
Yep, `fieldNamed` will only work where there is a `Columnar` in the nested structure of records. Since a `PrimaryKey` can be multi-column, it has to be itself a record which you nest into the referencing row record type.

@Fingerzam does that solve your issue?

**Comment by Fingerzam** [comment]
Yeah that works. Could this be explained in tutorial part2? The example with address lines could just be expanded to also rename the column for `_addressForUser` together with an explanation.

**Comment by amano-kenji** [comment]
I have a primary key that spans two columns. How can I use `fieldNamed` with such a key? Is there an example?
